src/channel.rs

```diff
@@ -88,6 +120,59 @@ impl<T> Clone for Sender<T> {
     }
 }
 
+impl<T> SyncSender<T> {
+    /// Sends a message to the corresponding receiver of this sender.
+    ///
+    /// This function will block until space in the internal buffer becomes available.
+    /// Otherwise, the message provided will be enqueued on the channel immediately, and
+    /// this function will return immediately.
+    ///
+    /// If an I/O error happens while sending the message, or if the receiver
+    /// has gone away, then an error will be returned. Note that I/O errors here
+    /// are generally quite abnormal.
+    pub fn send(&self, t: T) -> io::Result<()> {
+        self.tx.send(t).map_err(|e| {
+            match e {
+                channel::SendError::Io(e) => e,
+                channel::SendError::Disconnected(_) => {
+                    io::Error::new(io::ErrorKind::Other,
+                                   "channel has been disconnected")
+                }
+            }
+        })
+    }
+
+    /// Sends a message to the corresponding receiver of this sender.
+    ///
+    /// The message provided will be enqueued on the channel immediately, and
+    /// this function will return immediately.
+    ///
+    /// If an I/O error happens while sending the message, or if the receiver
+    /// has gone away, or the buffer is full, then an error will be returned.
+    /// Note that I/O errors here are generally quite abnormal.
+    pub fn try_send(&self, t: T) -> io::Result<()> {
+        self.tx.try_send(t).map_err(|e| {
+            match e {
+                channel::TrySendError::Io(e) => e,
+                channel::TrySendError::Disconnected(_) => {
+                    io::Error::new(io::ErrorKind::Other,
+                                   "channel has been disconnected")
+                }
+                channel::TrySendError::Full(_) => {
+                    io::Error::new(io::ErrorKind::Other,
+                                   "channel full")
```

Comment:
I think this error case may want to be surfaced in the type itself, it's a standard error case for `try_send` and may be difficult to look at otherwise.